azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -99,10 +99,11 @@
   @Inject
   public AzkabanExecutorServer(final Props props,
       final ExecutorLoader executionLoader,
-      final FlowRunnerManager runnerManager) throws Exception {
+      final FlowRunnerManager runnerManager, final MetricsManager metricsManager) throws Exception {
```

Comment:
Would it be possible to not touch `AzkabanExecutorServer` yet? This earlier PR has conflicting changes: https://github.com/azkaban/azkaban/pull/1113. Or, if you're going even further with injection on this class, maybe you want to check out that PR so that you don't have to figure out the same things again.